src/library/tactic/tactic_log.cpp

```diff
@@ -85,8 +91,8 @@ static tactic_log::summary summarize(tactic_state const & s) {
 void log_tactic(ast_id id, tactic_state const & before, tactic_state const & after, bool success) {
     if (auto log = get_tactic_log()) {
         lean_assert(!log->m_exported);
-        auto s1 = log->get_id(summarize(before));
-        auto s2 = is_eqp(before, after) ? s1 : log->get_id(summarize(after));
+        auto s1 = log->get_id(before, summarize(before));
+        auto s2 = is_eqp(before, after) ? s1 : log->get_id(after, summarize(after));
```

Comment:
Why not just call `summarize` in `get_id` if you are going to pass the tactic_state?